Sources/StructuralCore/Structural.swift

```diff
@@ -21,103 +74,187 @@ public protocol Structural {
     init(structuralRepresentation: StructuralRepresentation)
 
     /// A structural representation of `self`.
-    var structuralRepresentation: StructuralRepresentation { get set }
+    var structuralRepresentation: StructuralRepresentation {
+        get set
+    }
 }
 
-/// Structural representation of a Swift struct.
+/// The structural representation of a Swift struct type.
+///
+/// `StructuralStruct<...>` is the compiler-synthesized
+/// `StructuralRepresentation` for structs conforming to `Structural`.
+///
+/// - Parameter Properties: a type representing the names and values of stored
+///   properties.
 public struct StructuralStruct<Properties> {
-    public var type: Any.Type?
-    public var properties: Properties
+    /// The type of the represented struct type.
```

Comment:
Note: it is not easy to explain the case where `representedType` can be `nil`:
```
The type of the represented struct type, or `nil` if ...
```
This may be a design smell.

Reply:
Yep, as discussed earlier it might be desirable to address this. Similarly to the other comment, it's better to do it in a dedicated PR to validate changes over the uses of previous encoding that allowed for `nil` type.